### ISSUE TITLE
use `String` by default for `BigNumber` data

### ISSUE THREAD

**Issue by fuxingloh** [DeFiCh/whale#166]
#### What would you like to be added:

As Whale APIs are controlled by the jellyfish team, we should default all BigNumber to use `toFixed(8)` representation. This will not suffer from accidental precision loss and String is used. Although String might be awkward as it does not present number as the natural data type, it takes on the non-error approach as it forces the consumer to pick their own arbitrary number library for parsing numbers.

Hence, we need to change all number-sensitive data output into String as toFixed(8).
Before we do that we also need to fix https://github.com/DeFiCh/jellyfish/issues/392 and https://github.com/DeFiCh/jellyfish/issues/393

/area whale-api-client module-api
/triage accepted
/priority urgent-now
/assign @canonbrother

For TokenInfo

```diff
function mapTokenData (id: string, tokenInfo: TokenInfo): TokenData {
  return {
    id: id,
    symbol: tokenInfo.symbol,
    symbolKey: tokenInfo.symbolKey,
    name: tokenInfo.name,
    decimal: tokenInfo.decimal,
-    limit: tokenInfo.limit,
+    limit: tokenInfo.limit.toFixed(8),
    mintable: tokenInfo.mintable,
    tradeable: tokenInfo.tradeable,
    isDAT: tokenInfo.isDAT,
    isLPS: tokenInfo.isLPS,
    finalized: tokenInfo.finalized,
-    minted: tokenInfo.minted,    
+    minted: tokenInfo.minted.toFixed(8),
    creation: { tx: tokenInfo.creationTx, height: tokenInfo.creationHeight },
    destruction: { tx: tokenInfo.destructionTx, height: tokenInfo.destructionHeight },
    collateralAddress: tokenInfo.collateralAddress
  }
}
```

For PoolPair:

```ts
function mapPoolPair (id: string, poolPairInfo: PoolPairInfo): PoolPairData {
  return {
    id,
    symbol: poolPairInfo.symbol,
    name: poolPairInfo.name,
    status: poolPairInfo.status,
    tokenA: {
      id: poolPairInfo.idTokenA,
      reserve: poolPairInfo.reserveA.toFixed(8),
      blockCommission: poolPairInfo.blockCommissionA.toFixed(8)
    },
    tokenB: {
      id: poolPairInfo.idTokenB,
      reserve: poolPairInfo.reserveB.toFixed(8),
      blockCommission: poolPairInfo.blockCommissionB.toFixed(8)
    },
    commission: poolPairInfo.commission.toFixed(8),
    totalLiquidity: poolPairInfo.totalLiquidity.toFixed(8),
    tradeEnabled: poolPairInfo.tradeEnabled,
    ownerAddress: poolPairInfo.ownerAddress,
    rewardPct: poolPairInfo.rewardPct.toFixed(8),
    customRewards: poolPairInfo.customRewards?.toFixed(8),
    creation: {
      tx: poolPairInfo.creationTx,
      height: poolPairInfo.creationHeight.toNumber()
    }
  }
}
```

**Comment by fuxingloh** [comment]
The issues that I opened in jellyfish are related to this, which is ultimately related to Wallet. While developing the wallet, I found a few inconsistencies and things for improvement. They are `priority/urgent-now` as we should have them fixed before Wallet hit production.

**Comment by fuxingloh** [comment]
To implement as part of https://github.com/DeFiCh/whale/pull/167, you can use that PR or create one PR yourself.

**Comment by canonbrother** [comment]
yea i straight away fixed it at #167.

Will we have issue if we hardcode `toFixed(8)`? Diff tokens are diff decimails eg: USDT is 6, ETH is 18

**Comment by fuxingloh** [comment]
> yea i straight away fixed it at #167.
> 
> Will we have issue if we hardcode `toFixed(8)`? Diff tokens are diff decimals eg: USDT is 6, ETH is 18
> 
> ```
> tokenA: {
>       id: poolPairInfo.idTokenA,
>       reserve: poolPairInfo.reserveA.toFixed(8),
>       blockCommission: poolPairInfo.blockCommissionA.toFixed(8)  // tokenA could be ETH which has 18 decimals
> },
> tokenB: {
>       id: poolPairInfo.idTokenB,
>       reserve: poolPairInfo.reserveB.toFixed(8),
>       blockCommission: poolPairInfo.blockCommissionB.toFixed(8)
> },
> ```

Currently, the account system doesn't allow non 8 decimal token values to be used. But what you said it's important we should not hardcode that value. Yeah, we should use another function, maybe `toString()`?

**Comment by canonbrother** [comment]
> > yea i straight away fixed it at #167.
> > Will we have issue if we hardcode `toFixed(8)`? Diff tokens are diff decimals eg: USDT is 6, ETH is 18
> > ```
> > tokenA: {
> >       id: poolPairInfo.idTokenA,
> >       reserve: poolPairInfo.reserveA.toFixed(8),
> >       blockCommission: poolPairInfo.blockCommissionA.toFixed(8)  // tokenA could be ETH which has 18 decimals
> > },
> > tokenB: {
> >       id: poolPairInfo.idTokenB,
> >       reserve: poolPairInfo.reserveB.toFixed(8),
> >       blockCommission: poolPairInfo.blockCommissionB.toFixed(8)
> > },
> > ```
> 
> Currently, the account system doesn't allow non 8 decimal token values to be used. But what you said it's important we should not hardcode that value. Yeah, we should use another function, maybe `toString()`?

yea i think use `toString()` for first

**Comment by fuxingloh** [comment]
closed via #167
